zerver/tests/test_queue_worker.py

```diff
@@ -214,32 +217,82 @@ def cancel(self) -> None:
 
     def test_mirror_worker(self) -> None:
         fake_client = self.FakeClient()
+        stream = get_stream('Denmark', get_realm('zulip'))
+        stream_to_address = encode_email_address(stream)
         data = [
             dict(
                 message=u'\xf3test',
                 time=time.time(),
-                rcpt_to=self.example_email('hamlet'),
-            ),
-            dict(
-                message='\xf3test',
-                time=time.time(),
-                rcpt_to=self.example_email('hamlet'),
-            ),
-            dict(
-                message='test',
-                time=time.time(),
-                rcpt_to=self.example_email('hamlet'),
-            ),
-        ]
+                rcpt_to=stream_to_address
+            )
+        ] * 3
         for element in data:
             fake_client.queue.append(('email_mirror', element))
 
-        with patch('zerver.worker.queue_processors.mirror_email'):
+        with patch('zerver.worker.queue_processors.mirror_email') as mirror_email:
             with simulated_queue_client(lambda: fake_client):
                 worker = queue_processors.MirrorWorker()
                 worker.setup()
                 worker.start()
 
+            self.assertEqual(mirror_email.call_count, 3)
+
+    def test_mirror_worker_rate_limiting(self) -> None:
+        fake_client = self.FakeClient()
+        realm = get_realm('zulip')
+        clear_history(RateLimitedRealmMirror(realm))
+        stream = get_stream('Denmark', realm)
+        stream_to_address = encode_email_address(stream)
+        data = [
+            dict(
+                message=u'\xf3test',
+                time=time.time(),
+                rcpt_to=stream_to_address
+            )
+        ] * 5
+        for element in data:
+            fake_client.queue.append(('email_mirror', element))
+
+        with patch('zerver.worker.queue_processors.mirror_email') as mirror_email:
+            with self.settings(RATE_LIMITING_MIRROR_REALM_RULES=[(10, 2)]):
+                with simulated_queue_client(lambda: fake_client):
+                    start_time = time.time()
```

Comment:
I would actually mock `time.time()` to be some fixed value for the code below; it's better coding practice because it structurally avoids a test flake caused by slow hardware causing too much time to pass.